File: android/app/src/main/java/com/rork/porchivo/ui/screens/PaymentsScreen.kt
package com.rork.porchivo.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Apartment
import androidx.compose.material.icons.outlined.Description
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavController
import com.rork.porchivo.ui.components.EmptyState
import com.rork.porchivo.ui.theme.PorchivoTheme
import com.rork.porchivo.ui.viewmodel.AppViewModel

@Composable
fun PaymentsScreen(
    navController: NavController,
    modifier: Modifier = Modifier,
    appViewModel: AppViewModel = viewModel(),
) {
    val c = PorchivoTheme.colors
    val orgMembership by appViewModel.orgMembership.collectAsStateWithLifecycle()

    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(c.background),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(
            start = 16.dp, end = 16.dp, top = 16.dp, bottom = 24.dp,
        ),
        verticalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        item {
            Text(
                text = "Payments",
                color = c.textPrimary,
                fontSize = 26.sp,
                fontWeight = FontWeight.Black,
            )
        }
        item {
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(containerColor = c.surface),
            ) {
                Row(
                    modifier = Modifier.padding(12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                ) {
                    Box(
                        modifier = Modifier
                            .size(40.dp)
                            .background(c.accentSoft, CircleShape),
                        contentAlignment = Alignment.Center,
                    ) {
                        Icon(
                            imageVector = Icons.Outlined.Apartment,
                            contentDescription = null,
                            tint = c.accent,
                            modifier = Modifier.size(18.dp),
                        )
                    }
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            text = orgMembership?.orgName ?: "Your Community",
                            color = c.textPrimary,
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold,
                        )
                        Text(
                            text = "Resident",
                            color = c.textSecondary,
                            fontSize = 12.sp,
                        )
                    }
                }
            }
        }
        item {
            DuesCard()
        }
        item {
            Text(
                text = "Payment History",
                color = c.textPrimary,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
            )
        }
        item {
            EmptyState(
                icon = Icons.Outlined.Description,
                title = "No payments yet",
                body = "Your payment history and invoices will appear here once your community sets up billing.",
            )
        }
    }
}

@Composable
private fun DuesCard() {
    val c = PorchivoTheme.colors
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = c.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    text = "HOA Dues",
                    color = c.textPrimary,
                    fontSize = 17.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.weight(1f),
                )
                Text(
                    text = "Current",
                    color = c.success,
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier
                        .background(c.successSoft, RoundedCornerShape(8.dp))
                        .padding(horizontal = 8.dp, vertical = 4.dp),
                )
            }
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = "Next payment due",
                color = c.textSecondary,
                fontSize = 13.sp,
            )
            Text(
                text = "No upcoming dues",
                color = c.accent,
                fontSize = 20.sp,
                fontWeight = FontWeight.Black,
            )
        }
    }
}

File: ios/Porchivo/Views/PaymentsScreen.swift
//
//  PaymentsScreen.swift
//  Porchivo
//
//  Community tier tab — HOA dues, payment history, invoices.
//  Shown when user has an active org membership.
//

import SwiftUI

struct PaymentsScreen: View {
    @Environment(AppState.self) private var appState
    @Environment(\.porchivo) private var c
    @State private var path = NavigationPath()

    var body: some View {
        NavigationStack(path: $path) {
            ScrollView {
                VStack(spacing: 16) {
                    orgHeader
                    duesCard
                    historySection
                }
                .padding(.horizontal, 16)
                .padding(.top, 16)
                .padding(.bottom, 24)
            }
            .background(c.background.ignoresSafeArea())
            .navigationTitle("Payments")
        }
    }

    private var orgHeader: some View {
        HStack(spacing: 12) {
            ZStack {
                Circle().fill(c.accentSoft)
                Image(systemName: "building.2.fill")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundStyle(c.accent)
            }
            .frame(width: 40, height: 40)

            VStack(alignment: .leading, spacing: 2) {
                Text(appState.orgMembership?.orgName ?? "Your Community")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                Text("Resident")
                    .font(.system(size: 12))
                    .foregroundStyle(c.textSecondary)
            }
            Spacer()
        }
        .padding(12)
        .background(c.surface, in: .rect(cornerRadius: Radius.md))
    }

    private var duesCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("HOA Dues")
                    .font(.system(size: 17, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                Spacer()
                Pill(text: "Current", tint: c.success, softTint: c.successSoft)
            }
            Text("Next payment due")
                .font(.system(size: 13))
                .foregroundStyle(c.textSecondary)
            Text("No upcoming dues")
                .font(.system(size: 20, weight: .black))
                .foregroundStyle(c.accent)
        }
        .padding(16)
        .background(c.surface, in: .rect(cornerRadius: Radius.lg))
        .shadow(color: c.textPrimary.opacity(0.05), radius: 8, y: 3)
    }

    private var historySection: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("Payment History")
                .font(.system(size: 18, weight: .bold))
                .foregroundStyle(c.textPrimary)

            EmptyState(
                symbol: "doc.text.fill",
                title: "No payments yet",
                message: "Your payment history and invoices will appear here once your community sets up billing."
            )
        }
    }
}

#Preview {
    PaymentsScreen().environment(AppState())
}
